packages/client-react-streaming/src/ManualDataTransport/serialization.test.ts

```diff
@@ -0,0 +1,41 @@
+import test, { describe } from "node:test";
+import { revive, stringify } from "./serialization.js";
+import { outsideOf } from "../util/runInConditions.js";
+import { htmlEscapeJsonString } from "./htmlescape.js";
+import assert from "node:assert";
+
+describe(
+  "serialization and deserialization of data",
+  // we do not test the bundle, so we really just need to run this test in one environment
+  { skip: outsideOf("node") },
+  () => {
+    for (const [data, serialized] of [
+      [{ a: 1, b: 2, c: 3 }, '{"a":1,"b":2,"c":3}'],
+      [
+        { a: "$apollo.undefined$", b: 2, c: undefined },
+        '{"a":"$apollo.undefined$","b":2,"c":undefined}',
+      ],
+      [
+        { a: "a$apollo.undefined$", b: undefined, c: 3 },
+        '{"a":"a$apollo.undefined$","b":undefined,"c":3}',
+      ],
+      [
+        {
+          a: "$$apollo.undefined$",
+          b: 2,
+          c: undefined,
+          d: "$apollo.undefined$",
+        },
+        '{"a":"$$apollo.undefined$","b":2,"c":undefined,"d":"$apollo.undefined$"}',
+      ],
+      [{ a: undefined, b: 2, c: 3 }, '{"a":undefined,"b":2,"c":3}'],
+    ] as const) {
+      test(JSON.stringify(data), () => {
+        const stringified = stringify(data);
+        const result = revive(eval(`(${htmlEscapeJsonString(stringified)})`));
```

Comment:
I notice you're not using `eval` in the actual usage of `revive`. Should this try and test `revive` as well in the way that it is used in the other parts of the codebase? Perhaps I'm missing something, but this feels a bit like a false positive since the default `revive` just returns the value unchanged (which means `eval` is doing all the work here).

Reply:
That would probably mean writing it into a `<script>` tag in a file, starting a browser and then pointing it to that file ;)

=> Essentially, what the browser is doing is an `eval`, and I don't think we really have a better way of testing it.